### ISSUE TITLE
Unable to detect version on mercurial in a pre commit merge state

### ISSUE THREAD

**Issue by jaraco** [pypa/setuptools-scm#72]
With 1.10.0, setuptools_scm is failing to detect a version in one of my Hg projects:

```
$ python setup.py test
Traceback (most recent call last):
  File "setup.py", line 61, in <module>
    setuptools.setup(**setup_params)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/distutils/core.py", line 108, in setup
    _setup_distribution = dist = klass(attrs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/setuptools/dist.py", line 272, in __init__
    _Distribution.__init__(self,attrs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/distutils/dist.py", line 281, in __init__
    self.finalize_options()
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/setuptools/dist.py", line 327, in finalize_options
    ep.load()(self, ep.name, value)
  File "/Users/jaraco/Dropbox/code/yg/support/yg.jenkins/.eggs/setuptools_scm-1.10.0-py3.5.egg/setuptools_scm/integration.py", line 19, in version_keyword
  File "/Users/jaraco/Dropbox/code/yg/support/yg.jenkins/.eggs/setuptools_scm-1.10.0-py3.5.egg/setuptools_scm/__init__.py", line 102, in get_version
  File "/Users/jaraco/Dropbox/code/yg/support/yg.jenkins/.eggs/setuptools_scm-1.10.0-py3.5.egg/setuptools_scm/__init__.py", line 80, in _do_parse
LookupError: setuptools-scm was unable to detect version for '/Users/jaraco/Dropbox/code/yg/support/yg.jenkins'.

Make sure you're not using GitHub's tarballs (or similar ones), as those don't contain the necessary metadata. Use PyPI's tarballs instead.

For example, if you're using pip, instead of https://github.com/user/proj/archive/master.zip use git+https://github.com/user/proj.git#egg=proj
```

If I change the version to 1.9.x, it works again.

**Comment by jaraco** [comment]
Seems version detection is working in some cases:

```
$ PYTHONPATH=.eggs/setuptools_scm-1.10.0-py3.5.egg python
Python 3.5.1 (v3.5.1:37a07cee5969, Dec  5 2015, 21:12:44) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import setuptools_scm
>>> setuptools_scm.get_version()
'2.7.dev3+n5b8ca75b42a4'
```

**Comment by RonnyPfannschmidt** [comment]
is the code availiable somewhere? i suspect this is a regression introduced in the scm parse mechanism

**Comment by jaraco** [comment]
I'm beginning to think PEBCAK.

**Comment by jaraco** [comment]
The issue seems to be with the merge. Here's a repro against a clean Hg repo:

```
$ hg init setuptools_scm-72
$ cd setuptools_scm-72/
$ cat > setup.py
import setuptools
setuptools.setup(
  name='foo',
  use_scm_version=True,
  setup_requires=['setuptools_scm==1.9'],
)
$ hg add setup.py
$ hg ci -m "Initial commit"
$ hg update null
0 files updated, 0 files merged, 1 files removed, 0 files unresolved
$ touch foo
$ hg add foo
$ hg ci -m "Adding foo"
created new head
$ hg update 0
1 files updated, 0 files merged, 1 files removed, 0 files unresolved
$ hg merge 1
1 files updated, 0 files merged, 0 files removed, 0 files unresolved
(branch merge, don't forget to commit)
$ python setup.py --version

Installed /Users/jaraco/Dropbox/code/public/setuptools_scm/setuptools_scm-72/.eggs/setuptools_scm-1.9.0-py3.5.egg
0.0.0
$ sed -i -e 's/1.9/1.10/' setup.py
$ python setup.py --version

Installed /Users/jaraco/Dropbox/code/public/setuptools_scm/setuptools_scm-72/.eggs/setuptools_scm-1.10.0-py3.5.egg
Traceback (most recent call last):
  File "setup.py", line 5, in <module>
    setup_requires=['setuptools_scm==1.10'],
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/distutils/core.py", line 108, in setup
    _setup_distribution = dist = klass(attrs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/setuptools/dist.py", line 272, in __init__
    _Distribution.__init__(self,attrs)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/distutils/dist.py", line 281, in __init__
    self.finalize_options()
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/setuptools/dist.py", line 327, in finalize_options
    ep.load()(self, ep.name, value)
  File "/Users/jaraco/Dropbox/code/public/setuptools_scm/setuptools_scm-72/.eggs/setuptools_scm-1.10.0-py3.5.egg/setuptools_scm/integration.py", line 19, in version_keyword
  File "/Users/jaraco/Dropbox/code/public/setuptools_scm/setuptools_scm-72/.eggs/setuptools_scm-1.10.0-py3.5.egg/setuptools_scm/__init__.py", line 102, in get_version
  File "/Users/jaraco/Dropbox/code/public/setuptools_scm/setuptools_scm-72/.eggs/setuptools_scm-1.10.0-py3.5.egg/setuptools_scm/__init__.py", line 80, in _do_parse
LookupError: setuptools-scm was unable to detect version for '/Users/jaraco/Dropbox/code/public/setuptools_scm/setuptools_scm-72'.

Make sure you're not using GitHub's tarballs (or similar ones), as those don't contain the necessary metadata. Use PyPI's tarballs instead.

For example, if you're using pip, instead of https://github.com/user/proj/archive/master.zip use git+https://github.com/user/proj.git#egg=proj
$ hg ci -m "Finish merge"
$ python setup.py --version
0.1.dev3+na2e799fa4d39
```

**Comment by RonnyPfannschmidt** [comment]
there is a bug in the hg output template in the merge state

```
$ SETUPTOOLS_SCM_DEBUG=1 python -m setuptools_scm  
root '/home/ronny/foo'
looking for ep setuptools_scm.parse_scm /home/ronny/foo
found ep .hg = setuptools_scm.hg:parse
cmd 'hg id -i -t'
out '9040ec879713+11faf78c8d96+ tip\n'
tag tip
version <LegacyVersion('tip')>
cmd 'hg parents --template "{latesttag} {latesttagdistance}"'
out 'null 3null 3'
Traceback (most recent call last):
  File "/usr/lib64/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib64/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/ronny/foo/.env/lib/python2.7/site-packages/setuptools_scm/__main__.py", line 9, in <module>
    print('Guessed Version', get_version())
  File "/home/ronny/foo/.env/lib/python2.7/site-packages/setuptools_scm/__init__.py", line 102, in get_version
    version = _do_parse(root, parse)
  File "/home/ronny/foo/.env/lib/python2.7/site-packages/setuptools_scm/__init__.py", line 80, in _do_parse
    "use git+https://github.com/user/proj.git#egg=proj" % root)
LookupError: setuptools-scm was unable to detect version for '/home/ronny/foo'.

Make sure you're not using GitHub's tarballs (or similar ones), as those don't contain the necessary metadata. Use PyPI's tarballs instead.

For example, if you're using pip, instead of https://github.com/user/proj/archive/master.zip use git+https://github.com/user/proj.git#egg=proj
```

**Comment by RonnyPfannschmidt** [comment]
fixed in version 1.10.1

**Comment by jaraco** [comment]
Confirmed. Huge thanks!
